sbt-bridge/src/xsbt/CompilerInterface.java

```diff
@@ -6,36 +6,44 @@
 import xsbti.AnalysisCallback;
 import xsbti.Logger;
 import xsbti.Reporter;
-import xsbti.Severity;
 import xsbti.compile.*;
 
-import java.io.File;
-
-import dotty.tools.dotc.core.Contexts.ContextBase;
 import dotty.tools.dotc.Main;
-import dotty.tools.dotc.interfaces.*;
 
 import java.lang.reflect.InvocationTargetException;
-import java.net.URLClassLoader;
+import java.io.File;
 
 /**
  * The new compiler interface is [[dotty.tools.xsbt.CompilerBridge]] that extends the new `xsbti.CompilerInterface2`.
  * This interface is kept for compatibility with Mill and the sbt 1.3.x series.
  */
 public final class CompilerInterface {
-  public CachedCompiler newCompiler(String[] options, Output output, Logger initialLog, Reporter initialDelegate) {
-    // The classloader that sbt uses to load the compiler bridge is broken
-    // (see CompilerClassLoader#fixBridgeLoader for details). To workaround
-    // this we construct our own ClassLoader and then run the following code
-    // with it:
-    //   new CachedCompilerImpl(options, output)
+  public CachedCompiler newCompiler(String[] options, Output output, Logger initialLog, Reporter initialDelegate) throws java.lang.Exception {
+    if(isClassLoaderValid()) {
+      return new CachedCompilerImpl(options, output);
+    } else {
+      initialLog.warn(() -> "The compiler class loader is badly configured.");
```

Comment:
This message should mention the minimum version users should upgrade to for sbt/sbt-dotty/mill/bloop.